rs/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+use aggregator::init_zstd_encoder;
+use core::slice;
+use std::io::Write;
+use std::os::raw::{c_char, c_uchar};
+use std::ptr::null;
+
+fn out_as_err(err: &str, out: &mut [u8]) -> *const c_char {
+    let msg = if err.len() + 1 > out.len() {
+        "compress_scroll_batch_bytes: not enough output buffer for the error message"
+    } else {
+        err
+    };
+
+    let cpy_src = unsafe { slice::from_raw_parts(msg.as_ptr(), msg.len()) };
+    out[..cpy_src.len()].copy_from_slice(cpy_src);
+    out[cpy_src.len()] = 0; // build the c-style string
+    out.as_ptr() as *const c_char
+}
+
+/// Entry
+#[no_mangle]
+pub unsafe extern "C" fn compress_scroll_batch_bytes(
+    src: *const c_uchar,
+    src_size: u64,
+    output_buf: *mut c_uchar,
+    output_buf_size: *mut u64,
+) -> *const c_char {
+    let buf_size = *output_buf_size;
+    let src = unsafe { slice::from_raw_parts(src, src_size as usize) };
+    let out = unsafe { slice::from_raw_parts_mut(output_buf, buf_size as usize)};
+
+    let mut encoder = init_zstd_encoder();
```

Comment:
is it possible `init_zstd_encoder` once?

Reply:
nice catch. I tried with once_cell but failed. @noel2004 have we supported single initialization of the encoder? It's critical for performance.

Reply:
I think you can not simply reuse the `zstd_encoder`. According to the the underlying c code, this woould initialize a new context object and should be only used to compress single input bytes.

Have this really caused some impaction to the performance?

Reply:
it's ok if can't reuse it. just look here superficially, and think it will be better to reuse it.

Reply:
Yes. @noel2004  From the doc:
```
  When compressing many times,
  it is recommended to allocate a context just once,
  and re-use it for each successive compression operation.
  This will make workload friendlier for system's memory.
  Note : re-using context is just a speed / resource optimization.
         It doesn't change the compression ratio, which remains identical.
  Note 2 : In multi-threaded environments,
         use one different context per thread for parallel execution.
```
Initializing the context only once for each thread is recommended. Especially for Golang's memory management:
```
This will make workload friendlier for system's memory.
```

Reply:
closed this comment, but it's still a nice-to-have feature.